<?xml version="1.0"?>
<launch>
    <!-- 차선 검출 및 조향 제어 노드 -->
    <node name="onelane_detection" pkg="onelane_detection" type="onelane_detection.py" output="screen">
        <!-- 카메라 이미지 토픽 -->
        <param name="image_topic" value="/usb_cam/image_rect_color" />
        
        <!-- 제어 명령 발행 토픽 -->
        <param name="cmd_topic" value="/camera/steering" />
`
        <!-- 차선 선택: "left" 또는 "right" -->
        <param name="lane_side" value="right" />
        
        <!-- Sliding Window 알고리즘 파라미터 -->
        <param name="window_num" value="12" />
        <param name="window_margin" value="30" />
        <param name="crop_start_ratio" value="0.5" />
        
        <!-- HSV 색상 필터링 임계값 (노란색 차선 기준) -->
        <rosparam param="hsv_lower">[10, 50, 50]</rosparam>
        <rosparam param="hsv_upper">[40, 255, 255]</rosparam>
        
        <!-- 조향 기준점 비율 (0.0~1.0) -->
        <param name="steering_ratio" value="0.8" />
        <!-- 왼쪽/오른쪽 차선별 개별 설정 (옵션) -->
        <!-- <param name="steering_ratio_left" value="0.8" /> -->
        <!-- <param name="steering_ratio_right" value="0.8" /> -->
        
        <!-- 조향 게인 -->
        <!-- 통합 게인 (설정 시 좌우 구분 무시) -->
        <!-- <param name="steering_gain" value="0.6" /> -->
        <!-- 왼쪽/오른쪽 차선별 개별 게인 -->
        <param name="steering_gain_left" value="0.66" />
        <param name="steering_gain_right" value="0.55" />
        
        <!-- 주행 제어 -->
        <param name="enable_drive" value="true" />
        
        <!-- 디버그 창 표시 여부 -->
        <param name="show_debug" value="true" />
    </node>
</launch>
